Formula/ThorsSerializer.rb

```diff
@@ -0,0 +1,59 @@
+class Thorsserializer < Formula
+  desc "Declarative Serialization Library (Json/Yaml) for C++"
+  homepage "https://github.com/Loki-Astari/ThorsSerializer"
+  url "https://github.com/Loki-Astari/ThorsSerializer.git",
+    :tag => "1.5.7",
+    :revision => "c391b49a36c7432a3fadbbb3e602bba1f61d4078"
+
+  depends_on "libyaml"
+
+  def install
+    ENV["COV"] = "gcov"
+
+    system "./configure", "--disable-binary",
+                          "--disable-vera",
+                          "--with-thor-build-on-travis",
```

Comment:
Out of wild curiosity, what is this argument doing/impacting?

Reply:
--disable-binary:   There is an experimental binary serializer/deserialization in this lib.
                                I want to convert this to use a "standard" binary serializer;
                                something like googles protocol buffer.

                                It's not needed for the primary use case so disabled here.

--disable-vera:       Vera++ does code style checking.
                                If people want to contribute code back they need to make sure it
                                adheres to the style guide lines. But it requires a whole bunch of
                                other libraries to work. It is not needed to simply build the library
                                so disable it to remove dependencies required.

--with-thor-build-on-travis:   I included all my submodules via the `git@github.com/XXX`
                                                The travis-ci.org does not support this. So I added this flag
                                                to support automated builds on travis. Its useful here becuase
                                                neither does your build system. Internally it just converts the
                                                submodules to use `https://github.com/XXX`

Reply:
I'm confused how that works for submodules given we're long passed the point at which they're fetched by the time we hit configure

Reply:
In my initial pull request I had overridden the `update_submodules` in `GitHubGitDownloadStrategy` to try and use `git@github.com/XXX` and it worked for the normal build. 

````
class GitWithGitAt < GitHubGitDownloadStrategy
  def update_submodules
````

But it failed when building the binaries.  

It would be nice if you supported `git@github.com/XXX` submodules as it makes working with submodules easier (I don't need to check them out separately to update them). You would need to create a SSH deployment key that people could add to their github repositories.